src/finchlite/codegen/c_codegen.py

```diff
@@ -604,12 +605,24 @@ def __init__(
         self.fptr = fptr
         self.types = types
         self.slots = slots
+        self.datastructures: dict[Hashable, Any] = {}
 
     def add_header(self, header):
         if header not in self._headerset:
             self.headers.append(header)
             self._headerset.add(header)
 
+    def add_datastructure(self, key: Hashable, handler: "Callable[[CContext], Any]"):
```

Comment:
Instead of `"Callable[[CContext], Any]"` you can `from __future__ import annotations` [ref](https://docs.python.org/3/library/__future__.html#module-contents) and have `Callable[[CContext], Any]`.